include/ibamr/IBFESurfaceMethod.h

```diff
@@ -111,7 +120,14 @@ class IBFESurfaceMethod : public IBStrategy
     static const std::string NORMAL_VELOCITY_SYSTEM_NAME;
     static const std::string PRESSURE_JUMP_SYSTEM_NAME;
     static const std::string TANGENTIAL_VELOCITY_SYSTEM_NAME;
+    static const std::array<std::string, NDIM> VELOCITY_JUMP_SYSTEM_NAME;
     static const std::string VELOCITY_SYSTEM_NAME;
+    static const std::string WSS_SYSTEM_NAME;
+    static const std::string P_SYSTEM_NAME;
+    static const std::string TAU_SYSTEM_NAME;
+
+    SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > mask_var;
+    int mask_current_idx, mask_new_idx, mask_scratch_idx;
```

Comment:
```suggestion
    int mask_current_idx = IBTK::invalid_index, mask_new_idx = IBTK::invalid_index, mask_scratch_idx = IBTK::invalid_index;
```
Are these actually used anywhere?

Reply:
Yup I'm using this Eulerian mask variable for the calculation of the pressure in `calculateInterfacialFluidForces`. Maybe there are other ways to go about this too?

Reply:
It does not look like that variable is actually being used as a masking variable (i.e. it is not being used to mark locations on the Cartesian grid). It looks like you are using it as a temporary copy of the pressure. I would suggest cloning the pressure scratch index. There is some infrastructure to do this and automatically setup the data allocation using a new class, `SAMRAIDataCache`.

Please switch over to using `SAMRAIDataCache` and get rid of the mask variable and indices.

Reply:
How can I clone the pressure scratch index and use wider ghost width? Don't I need to create and maintain a wider width variable myself? When I tried doing what you recommended here I get the following error:

```
P=00000:hier::VariableDatabase<DIM>::registerVariableAndContext error ...
P=00000:Attempting to to register variable ::P and context INSStaggeredHierarchyIntegrator::SCRATCH with ghost width = (4,4)
P=00000: This variable-context pair is already registered with a different ghost width. 
P=00000:
Abort trap: 6
```

Reply:
I think that was the reason I used the mask variable with a larger ghost width (6) needed for the calculation of the pressure in the traction. @abarret do you know a way to clone the pressure scratch index with a wider ghost width than 1 and avoid introducing a mask variable?

Reply:
Oh, OK --- in that case, probably the easiest thing to do will be to register a `SCRATCH` variable to store the ghost data.

Please change the name from `mask` to `p` and get rid of the `current` and `new` indices --- just set up a new `scratch` index.